tools/bisection/run-cpp-smoke.sh

```
@@ -50,6 +59,17 @@ g++ -std=c++17 -Wall -Wextra -Werror -O2 \
     -lpinyin_capi \
     -o "$BUILD_DIR/cpp-smoke"
 
+needed=$(readelf -d "$BUILD_DIR/cpp-smoke" | sed -n 's/.*Shared library: \[\(libpinyin[^]]*\)\].*/\1/p')
+if [ "$needed" != "libpinyin.so.15" ]; then
+    echo "fatal: smoke binary needs '$needed', expected libpinyin.so.15"
+    exit 1
+fi
+resolved=$(LD_LIBRARY_PATH= ldd "$BUILD_DIR/cpp-smoke" | sed -n 's/.*libpinyin\.so\.15 => \([^ ]*\).*/\1/p')
```

Shellcheck (0.11.0)

[warning] Remove space after = if trying to assign a value (for empty string, use var='' ... ).

(SC1007)